pkg/lib/database/filter.go

```diff
@@ -19,13 +19,26 @@ var (
 	ErrValueMustBeListSupporting = errors.New("value must be list supporting")
 )
 
-func MakeFiltersAnd(protoFilters []*model.BlockContentDataviewFilter, store ObjectStore) (FiltersAnd, error) {
+func MakeFilters(protoFilters []*model.BlockContentDataviewFilter, store ObjectStore) (Filter, error) {
 	if store == nil {
 		return FiltersAnd{}, fmt.Errorf("objectStore dependency is nil")
 	}
+	if len(protoFilters) == 0 {
+		return FiltersAnd{}, nil
+	}
 	spaceID := getSpaceIDFromFilters(protoFilters)
-	protoFilters = TransformQuickOption(protoFilters, nil)
+	operator := protoFilters[0].Operator
+	if operator == model.BlockContentDataviewFilter_No {
+		return makeFilterNone(protoFilters, store, spaceID)
```

Comment:
shouldn't it be makeFilterAnd?

Reply:
I can make this filter and expect it should work
```
	filters := []*model.BlockContentDataviewFilter{
		{
			RelationKey: "key1",
			Condition:   model.BlockContentDataviewFilter_Equal,
			Value:       pbtypes.Bool(true),
		},
		{
			Operator: model.BlockContentDataviewFilter_Or,
			NestedFilters: []*model.BlockContentDataviewFilter{
				{
					RelationKey: "key2",
					Condition:   model.BlockContentDataviewFilter_Equal,
					Value:       pbtypes.Bool(true),
				},
			},
		},
	}
```

Reply:
cases for recusrion:
f = `f1 AND f2 AND f3 AND ...`        - Op = None
f = `(f1 AND f2 AND f3 AND ...)`     - Op = And, parenthesis means Nested filters
f = `(f1 OR f2 OR f3 OR ...)`              - Op = Or, , parenthesis means Nested filters

Reply:
Also don't understand this logic.
Here we should always wrap these filters as AND filter.

Reply:
Made code cleaner and add test for @deff7 case